src/node_crypto.cc

```diff
@@ -1025,6 +1066,46 @@ void SecureContext::LoadPKCS12(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+#ifndef OPENSSL_NO_ENGINE
+void SecureContext::SetClientCertEngine(
+  const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  CHECK(args.Length() == 1 && args[0]->IsString());
+
+  SecureContext* sc = Unwrap<SecureContext>(args.This());
+
+  ClearErrorOnReturn clear_error_on_return;
+
+  // If an engine was previously set, free it.
+  // This is because SSL_CTX_set_client_cert_engine does not itself
+  // 'support' multiple calls by cleaning up before overwriting the
+  // client_cert_engine internal context variable
+  if (sc->ctx_->client_cert_engine != NULL) {
```

Comment:
I don't know... from an API point of view I think it would violate the 'principle of least surprise' if it threw the second time you called it (given the name of the function). I may not be able to come up with an example of changing the engine during the lifetime but maybe someone needs to set an engine on an object outside his own life-cycle-control? That guy would be annoyed! With the above work-around for missing cleanup in OpenSSL the API behaves as one would expect?